addons/mail/models/mail_message.py

```diff
@@ -674,11 +674,18 @@ def export_data(self, fields_to_export):
 
         return super(Message, self).export_data(fields_to_export)
 
-    def _update_content(self, body):
+    def _update_content(self, body, attachment_ids):
         self.ensure_one()
         thread = self.env[self.model].browse(self.res_id)
         thread._check_can_update_message_content(self)
         self.body = body
+        message_values = {
+            'model': self.model,
+            'body': body,
+            'res_id': self.res_id,
+        }
+        attachement_values = self.env['mail.thread']._message_post_process_attachments([], attachment_ids, message_values)
```

Comment:
`_message_post_process_attachments` should be called on the actual `thread` of the message, not as a static method.